File: problems/c/cprnummer/cprnummer.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    cpr = [int(val) for val in input() if val.isdigit()]
    list_values = [4, 3, 2, 7, 6, 5, 4, 3, 2, 1]
    cpr_sum = sum([a * b for a, b in zip(cpr, list_values)])
    if cpr_sum % 11 == 0:
        print(1)
    else:
        print(0)


if __name__ == '__main__':
    main()

File: problems/c/cprnummer/cprnummer.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    string cpr;
    string corresponding = "432765-4321";
    int cpr_sum = 0;

    cin >> cpr;

    for (size_t i=0; i<cpr.size(); i++) {
        if (cpr[i] == '-') {
            continue;
        }
        cpr_sum += (cpr[i] - 48) * (corresponding[i] - 48);
    }
    
    if (cpr_sum % 11 == 0) {
        cout << 1 << endl;
    }
    else {
        cout << 0 << endl;
    }
}